sdk/search/Azure.Search.Documents/src/Batching/ManualRetryDelay.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Search.Documents.Batching
+{
+    /// <summary>
+    /// Implements a "manual" version of an exponential back off retry policy
+    /// that allows us to block before sending the next request instead of
+    /// immediately after we get a response.
+    /// </summary>
+    internal class ManualRetryDelay
+    {
+        /// <summary>
+        /// The initial delay we use for calculating back off.
+        /// </summary>
+        public TimeSpan Delay { get; set; } = TimeSpan.FromSeconds(0.8);
+
+        /// <summary>
+        /// The maximum delay between attempts.
+        /// </summary>
+        public TimeSpan MaxDelay { get; set; } = TimeSpan.FromMinutes(1);
+
+        /// <summary>
+        /// Randomize with jitter.
+        /// </summary>
+        private readonly Random _jitter = new Random();
```

Comment:
Do you need this Random to be thread-safe?

Reply:
No, this is only used from the single consumer.